app/src/main/java/com/concordium/wallet/ui/cis2/retrofit/MetadataApiInstance.kt

```diff
@@ -2,42 +2,73 @@ package com.concordium.wallet.ui.cis2.retrofit
 
 import com.concordium.wallet.data.model.TokenMetadata
 import com.concordium.wallet.util.Log
+import com.concordium.wallet.core.backend.BackendError
+import com.concordium.wallet.core.backend.BackendErrorException
 import okhttp3.OkHttpClient
+import okhttp3.Interceptor
 import okhttp3.logging.HttpLoggingInterceptor
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 
+import java.security.MessageDigest
+import java.util.Base64
+import java.io.IOException
+
+fun sha256(input: ByteArray): String {
```

Comment:
This belongs in a utility class somewhere. I am a bit surprised that there is no existing sha256 function being used.

Reply:
Which lead to me realizing that wCCD does not use checksum, and when I tested it with an actual checksum-using token, I learned that I was consuming the response body in the interceptor :upside_down_face: